pyne/endf.pyx

```diff
@@ -152,18 +162,14 @@ class Library(rxdata.RxLib):
         # get mat_id
         line = fh.readline()
         mat_id = int(line[66:70].strip() or -1)
-        # store position of read
-        pos = fh.tell()
         # check for isomer (LIS0/LISO entry)
-        matflagstring = line + fh.read(3*self.line_length)
+        matflagstring = line + fh.read(3*len(line))
```

Comment:
Is there a reason to not use the member variable here?  The purpose of this variable is to be robust across Window vs Mac vs Linux.

Reply:
Well... I'm trying to use pyne.endf to read JENDL4 files.  Unfortunately, some of which are with '\r\n' and others have '\n', and the reading failed for files w/ '\r\n'.  So I followed the function of Library._read_headers() and found that for the files w/ '\r\n' the fh.read(3*self.line_length) goes too further!  I think that it is because of the universal newline mode; by experimenting, it seems to me that the size argument of read() is not that of file but the characters it returns, although I'm not quite sure...

Reply:
##### Sr087.dat: ASCII text
_read_tpid: chars_til_now =  81
_read_tpid: line_length   =  81
_read_tpid: offset        =  0
_read_tpid: len(line)     =  81
##### Sr088.dat: ASCII text, with CRLF line terminators
_read_tpid: chars_til_now =  82
_read_tpid: line_length   =  82
_read_tpid: offset        =  0
_read_tpid: len(line)     =  81


This way, regardless of the newline characters (`'\r\n'` or `'\n'`) and the value of `self.line_length`, `len(line)` always returns 81, and `self.offset` is always 0.

`self.chars_til_now = len(line) + self.line_length - 81`
`self.offset = self.line_length - self.chars_til_now`

Reply:
It seems to me that the same thing (reading too much characters) occurs in endf.Library._read_nucmfmt(), and I'm wondering how to fix them...